nncf/torch/quantization/algo.py

```diff
@@ -450,7 +453,10 @@ def __init__(self, config, should_init: bool = True):
         # noise on model evaluation (e.g. in AutoQ)
         self._should_setup_adjust_pad_ops = True
 
-        hw_config_type = self.config.get("hw_config_type")
+        hw_config_type = None
+        target_device = self.config.get('target_device', 'ANY')
+        if target_device != 'TRIAL':
+            hw_config_type = HWConfigType.from_str(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[target_device])
```

Comment:
Do you consider creating builder specific method like `builder.extract_algo_specific_config`, which extract compression config section and then extend it with builder specific attributes and structures like `target_device` or hw config? It would be clearer if builder will compose fully specified config in the base init and then use only it (only self._algo_config, for example). Now builder uses self.config and self._algo_config simultaneously which is a bit complicated I think

Reply:
@daniil-lyakhov you are right and this is an improvement for the future.